### DISCUSSION TITLE
DesiredSize property changed

### DISCUSSION THREAD

**Discussion by PapLabros** [AvaloniaUI/Avalonia#13078] [Q&A] [answered]
DesiredSize appears to be a DirectProperty defined in the Layoutable class but I can not use it in a binding because it seems that when its value is updated, the necessary events are not raised. The same applies when using the PropertyChangedEvent of the AvaloniaObject class, meaning that when the DesiredSize property is updated the PropertyChangedEvent is not raised. Is this behavior expected ?, and if so, is there another way to capture when the value of the DesiredSize property changes?

**Comment by PapLabros** [comment]
After further inspection, I noticed that the DesiredSize property doesn't use the SetAndRaise method when setting its value. That's the reason why the events are not triggered. Is there a reason for this design approach? 🤔 Thanks in advance!

**Comment by maxkatz6** [comment, marked as answer]
What exactly are you trying to do? And why wouldn't you use Bounds property and/or SizeChanged event?

**Reply by appel1** [reply]
Maybe there's a better way, but I wanted to create a MultiBinding on RenderTransform that creates a ScaleTransform that scales the control to fit. Didn't work very well when DesiredSize changes does not trigger the binding to re-evaluate.

**Comment by rabbitism** [comment]
I think DesiredSize is something used for passing information between Measure and Arrange layout calculation. Why do you need this?

**Comment by PapLabros** [comment]
My mistake, I was creating a component that had to calculate its size based on an aspect ratio but a binding was off leading me into a weird rabbit hole. Using the Bounds property and the SizeChanged event is indeed the solution I needed. Now for educational purposes only, isn't my consern relating to the usage of the SetAndRaise method valid?, or is there something I haven't quite understood relating to the StyledProperties?
